src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ConcurrencyLimiter.cs

```diff
@@ -38,7 +38,26 @@ public sealed class ConcurrencyLimiter : RateLimiter
         /// <param name="options">Options to specify the behavior of the <see cref="ConcurrencyLimiter"/>.</param>
         public ConcurrencyLimiter(ConcurrencyLimiterOptions options)
         {
-            _options = options ?? throw new ArgumentNullException(nameof(options));
+            if (options is null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+            if (options.PermitLimit <= 0)
+            {
+                throw new ArgumentException($"{nameof(options.PermitLimit)} must be set to a value greater than 0.");
```

Comment:
Use the overload that takes the argument name too here and in the block below.

If you used ArgumentOutIfRangeException instead, you could also include the invalid value.